lib/controller/page/type/page.summary/page.summary.controller.unit.spec.js

```diff
@@ -163,89 +164,109 @@ test('When all previous required questions have been answered - and a url is ret
   getInstancePropertyStub.withArgs('service', 'attachUserSubmission').returns(false)
   const pageSummaryController = proxyquire('./page.summary.controller', {
     '../../../../page/check-submits/check-submits': {checkSubmits: checkSubmitsStub},
     '../../../../presenter/pdf-payload': {pdfPayload: pdfPayloadStub},
-    '../../../../middleware/routes-output/submission-data-with-labels': {submissionDataWithLabels: submissionDataWithLabelsStub}
+    '../../../../middleware/routes-output/submission-data-with-labels': {submissionDataWithLabels: submissionDataWithLabelsStub},
+    '../../../../constants/constants': {
+      SERVICE_OUTPUT_EMAIL: 'bob@gov.uk'
+    }
   })
 
-  await pageSummaryController.postValidation({}, userData)
+  const userdata = getUserDataMethods({input: {email: 'test@emample.com'}})
+
+  await pageSummaryController.postValidation({}, userdata)
 
   const submissions = submitterClientSpy.getCall(0).args[0].submissions
 
-  t.equals(submissions[0].recipientType, 'user')
-  t.equals(submissions[0].attachments.length, 0)
+  t.equals(submissions[1].recipientType, 'user')
+  t.equals(submissions[1].attachments.length, 0)
   submitterClientSpy.resetHistory()
   t.end()
 })
 
-test.skip('it attaches json to submission when env var present', async t => {
-  getUserDataPropertyStub.resetHistory()
-
+test('it attaches json to submission when env var present', async t => {
   const pageSummaryController = proxyquire('./page.summary.controller', {
     '../../../../page/check-submits/check-submits': {checkSubmits: checkSubmitsStub},
     '../../../../presenter/pdf-payload': {pdfPayload: pdfPayloadStub},
     '../../../../middleware/routes-output/submission-data-with-labels': {submissionDataWithLabels: submissionDataWithLabelsStub},
     '../../../../constants/constants': {
       SERVICE_OUTPUT_JSON_ENDPOINT: 'https://example.com/adaptor',
       SERVICE_OUTPUT_JSON_KEY: 'shared_key',
-      RUNNER_URL: 'http://service-slug.formbuilder-services-test-dev:3000'
+      RUNNER_URL: 'http://service-slug.formbuilder-services-test-dev:3000',
+      SERVICE_OUTPUT_EMAIL: 'bob@gov.uk'
     }
   })
 
+  const userData = getUserDataMethods({input: {mock_question_1: 'mock_answer_1'}})
+
   await pageSummaryController.postValidation({}, userData)
 
   const submissions = submitterClientSpy.getCall(0).args[0].submissions
   t.equals(submissions[1].type, 'json')
   t.equals(submissions[1].url, 'https://example.com/adaptor')
   t.equals(submissions[1].encryption_key, 'shared_key')
-  t.deepEquals(submissions[1].user_answers, {
-    mock_question_1: 'mock_answer_1'
-  })
+  t.equals(submissions[1].user_answers.mock_question_1, 'mock_answer_1')
+
   t.deepEquals(submissions[1].attachments, [])
   submitterClientSpy.resetHistory()
   t.end()
 })
 
-test.skip('when there are files it attaches them to json submission', async t => {
-  getUserDataPropertyStub.resetHistory()
-
+test('when there are files it attaches them to json submission', async t => {
```

Comment:
This description doesn't seem to match the assertion, unless I'm missing something?

Reply:
Theres a few assertions, it matches one of them, which is probably the most important one.